// @getzep/zep-cloud/dist/esm/serialization/resources/graph/client/requests/AddTripleRequest.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
import * as core from "../../../../../core/index.mjs";
export const AddTripleRequest = core.serialization.object({
    createdAt: core.serialization.property("created_at", core.serialization.string().optional()),
    edgeAttributes: core.serialization.property("edge_attributes", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    expiredAt: core.serialization.property("expired_at", core.serialization.string().optional()),
    fact: core.serialization.string(),
    factName: core.serialization.property("fact_name", core.serialization.string()),
    factUuid: core.serialization.property("fact_uuid", core.serialization.string().optional()),
    graphId: core.serialization.property("graph_id", core.serialization.string().optional()),
    invalidAt: core.serialization.property("invalid_at", core.serialization.string().optional()),
    sourceNodeAttributes: core.serialization.property("source_node_attributes", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    sourceNodeLabels: core.serialization.property("source_node_labels", core.serialization.list(core.serialization.string()).optional()),
    sourceNodeName: core.serialization.property("source_node_name", core.serialization.string().optional()),
    sourceNodeSummary: core.serialization.property("source_node_summary", core.serialization.string().optional()),
    sourceNodeUuid: core.serialization.property("source_node_uuid", core.serialization.string().optional()),
    targetNodeAttributes: core.serialization.property("target_node_attributes", core.serialization.record(core.serialization.string(), core.serialization.unknown()).optional()),
    targetNodeLabels: core.serialization.property("target_node_labels", core.serialization.list(core.serialization.string()).optional()),
    targetNodeName: core.serialization.property("target_node_name", core.serialization.string().optional()),
    targetNodeSummary: core.serialization.property("target_node_summary", core.serialization.string().optional()),
    targetNodeUuid: core.serialization.property("target_node_uuid", core.serialization.string().optional()),
    userId: core.serialization.property("user_id", core.serialization.string().optional()),
    validAt: core.serialization.property("valid_at", core.serialization.string().optional()),
});
