// @getzep/zep-cloud/dist/esm/serialization/resources/graph/client/requests/DetectPatternsRequest.mjs
/**
 * This file was auto-generated by Fern from our API Definition.
 */
import * as core from "../../../../../core/index.mjs";
import { DetectConfig } from "../../../../types/DetectConfig.mjs";
import { RecencyWeight } from "../../../../types/RecencyWeight.mjs";
import { SearchFilters } from "../../../../types/SearchFilters.mjs";
import { PatternSeeds } from "../../../../types/PatternSeeds.mjs";
export const DetectPatternsRequest = core.serialization.object({
    detect: DetectConfig.optional(),
    edgeLimit: core.serialization.property("edge_limit", core.serialization.number().optional()),
    graphId: core.serialization.property("graph_id", core.serialization.string().optional()),
    limit: core.serialization.number().optional(),
    minOccurrences: core.serialization.property("min_occurrences", core.serialization.number().optional()),
    query: core.serialization.string().optional(),
    queryLimit: core.serialization.property("query_limit", core.serialization.number().optional()),
    recencyWeight: core.serialization.property("recency_weight", RecencyWeight.optional()),
    searchFilters: core.serialization.property("search_filters", SearchFilters.optional()),
    seeds: PatternSeeds.optional(),
    userId: core.serialization.property("user_id", core.serialization.string().optional()),
});
